packages/pwa-buildpack/lib/WebpackTools/configureWebpack.js

```diff
@@ -291,26 +306,45 @@ async function configureWebpack({ context, vendor = [], special = {}, env }) {
             hints: 'warning'
         };
         config.devtool = false;
-        config.optimization.minimizer = [
-            new TerserPlugin({
-                parallel: true,
-                cache: true,
-                terserOptions: {
-                    ecma: 8,
-                    parse: {
-                        ecma: 8
-                    },
-                    compress: {
-                        drop_console: true
-                    },
-                    output: {
-                        ecma: 7,
-                        semicolons: false
-                    },
-                    keep_fnames: true
+        config.optimization = {
+            ...config.optimization,
+            moduleIds: 'hashed',
+            runtimeChunk: 'single',
+            splitChunks: {
+                cacheGroups: {
+                    vendor: {
+                        test: new RegExp(vendorTest),
+                        name: 'vendors',
+                        chunks: 'all'
+                    }
                 }
+            },
+            minimizer: [
+                new TerserPlugin({
+                    parallel: true,
+                    cache: true,
+                    terserOptions: {
+                        ecma: 8,
+                        parse: {
+                            ecma: 8
+                        },
+                        compress: {
+                            drop_console: true
+                        },
+                        output: {
+                            ecma: 7,
+                            semicolons: false
+                        },
+                        keep_fnames: true
+                    }
+                })
+            ]
+        };
+        config.plugins.push(
+            new webpack.optimize.LimitChunkCountPlugin({
+                maxChunks: 5
```

Comment:
Does this mean we will only ever split into five chunks?

If so, can you explain why we would want this? Seems pretty arbitrary at the moment.

Reply:
Yeah, creating too many chunks can create HTTP overhead issues at the same time creating less number of them might create issues related to bundle size and caching. That is why the online community feels 5 is an ideal number. Also if we decide on doing the route based bundling, we will have to change this. This is a filler for now. If I remove this, webpack is creating 8-10 chunks on avg and that will hamper the app performance.